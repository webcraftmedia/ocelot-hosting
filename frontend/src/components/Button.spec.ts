import { renderSuspended } from '@nuxt/test-utils/runtime'
import Component from './Button.vue'
import { describe, expect, it } from 'vitest'

describe('Button', () => {
  describe('renders', async () => {
    it('primary', async () => {
      const html = await (
        await renderSuspended(Component, { props: { text: 'Primary Button' } })
      ).html()
      expect(html).toMatchSnapshot()
    })
    it('primary', async () => {
      const html = await (
        await renderSuspended(Component, { props: { type: 'secondary', text: 'Secondary Button' } })
      ).html()
      expect(html).toMatchSnapshot()
    })
  })
})
