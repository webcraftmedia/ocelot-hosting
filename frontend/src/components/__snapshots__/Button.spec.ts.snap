// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders > primary 1`] = `"<div id="test-wrapper"><button data-v-b534b83e="" type="button" class="primary">Primary Button</button></div>"`;

exports[`Button > renders > primary 2`] = `"<div id="test-wrapper"><button data-v-b534b83e="" type="button" class="secondary">Secondary Button</button></div>"`;
